spinalcordtoolbox/process_seg.py

```diff
@@ -254,3 +423,496 @@ def _find_AP_and_RL_diameter(major_axis, minor_axis, orientation, dim):
+    # object rather than around the origin of the coordinate system.
+    Xr = Xr + x0
+    Yr = Yr + y0
+
+    # Create coordinate mapping for interpolation
+    coords = np.column_stack([np.ravel(Yr), np.ravel(Xr)])
+
+    # Apply transformation
+    seg_crop_r_rotated = map_coordinates(seg_crop_r,
+                                         [coords[:, 0], coords[:, 1]],
+                                         order=1).reshape(seg_crop_r.shape)      # order of the spline interpolation --> order=1: linear interpolation
+
+    return seg_crop_r_rotated
+
+
+def _measure_rotated_diameters(seg_crop_r, seg_crop_r_rotated, dim, angle, iz, properties, verbose):
+    """
+    Measure the AP and RL diameters in the rotated segmentation.
+    This function counts the number of pixels along the AP and RL axes in the rotated segmentation and converts them
+    to physical dimensions using the provided pixel size.
+
+    :param seg_crop_r: Original cropped segmentation (used only for plotting).
+    :param seg_crop_r_rotated: Rotated segmentation (after applying angle) used to measure diameters. seg.shape[0] --> RL; seg.shape[1] --> PA
+    :param dim: Physical dimensions of the segmentation (in mm). X,Y respectively correspond to RL,PA.
+    :param angle: Rotation angle in radians (HOG angle found from the image)
+    :param iz: Integer slice number (z index) of the segmentation. Used for plotting purposes.
+    :param properties: Dictionary containing the properties of the original (not-rotated) segmentation. Used for plotting purposes.
+    :param verbose: Verbosity level. If 2, debug figures are created.
+    :return result: Dictionary containing the measured diameters and pixel counts along AP and RL axes.
+    """
+    # Get center of mass (which is computed in the PCA function); centermass_src: [RL, AP] (assuming RPI orientation)
+    # Note: I'm using [rl0, ap0] instead of [y0, x0] to make it easier to track the axes as numpy handle them in a bit unintuitive way :-D
+    rotated_bin = np.array(seg_crop_r_rotated > 0.5, dtype='uint8')  # binarize the rotated segmentation for PCA
+    # Check if segmentation is empty after rotation
+    if np.all(rotated_bin < NEAR_ZERO_THRESHOLD):
+        logging.debug('The rotated segmentation is empty.')
+        return {
+            'ap_pixel_count': np.nan,
+            'diameter_AP': np.nan,
+        }
+    else:
+        _, _, [rl0, ap0] = compute_pca(rotated_bin)    # same as `y0, x0 = region.centroid`
+        rl0_r, ap0_r = round(rl0), round(ap0)
+
+        # Note: seg_crop_r_rotated is soft (due to the rotation) so we sum its values to account for softness
+        # Sum non-zero pixels along AP axis, i.e., the number of pixels in the row corresponding to the center of mass along the RL axis
+        # Compute AP diameter average acrross 3 mm extent centered at rl0_r
+
+        # Use centroid for AP diameter
+        extent_avg = 30  # extent used for averaging the minimum (to account for noise)
```

Comment:
```suggestion
        extent_avg = 30  # extent of 3 mm (because seg was resampled to 0.1mm) used for averaging the minimum (to account for noise)
```